authentication.js

```diff
@@ -1,30 +1,120 @@
+  });
+};
+
+
+const testAuth = (z , bundle ) => {
   // Normally you want to make a request to an endpoint that is either specifically designed to test auth, or one that
   // every user will have access to, such as an account or profile endpoint like /me.
-  // In this example, we'll hit httpbin, which validates the Authorization Header against the arguments passed in the URL path
+  const promise = z.request({
+    method: 'GET',
+    url: bundle.authData.baseUrl+'/api/contacts?limit=1&minimal=1',
+    headers: { Authorization: 'Bearer '+bundle.authData.access_token },
+  });
 
   // This method can return any truthy value to indicate the credentials are valid.
   // Raise an error to show
-  return z.request({
-      url: bundle.authData.baseUrl+'/api/contacts?limit=1&minimal=1',
-    }).then((response) => {
+  return promise.then(response => {
 
-      if (typeof response.json === 'undefined') {
-        throw new Error('The URL you provided ('+bundle.authData.baseUrl+') is not the base URL of a Mautic instance');
-      }
+    if (typeof response.json === 'undefined') {
+      throw new Error('The URL you provided ('+bundle.inputData.baseUrl+') is not the base URL of a Mautic instance');
+    }
 
-      return response;
-    });
+    return response;
+  });
 };
 
+
 module.exports = {
-  type: 'basic',
+  type: 'oauth2',
   fields: [
+    {key: 'clientId', type: 'string', required: true, helpText: 'Your Client ID (Public Key) is available in Mautic > Settings > API Credentials > OAuth 2'},
+    {key: 'clientSecret', type: 'string', required: true, helpText: 'Your Client Secret (Secret Key) is available in Mautic > Settings > API Credentials > OAuth 2'},
     {key: 'baseUrl', type: 'string', required: true, helpText: 'The root URL of your Mautic installation starting with https://. E.g. https://my.mautic.net.'}
   ],
-  connectionLabel: '{{bundle.authData.username}}',
-
-  // The test method allows Zapier to verify that the credentials a user provides are valid. We'll execute this
-  // method whenver a user connects their account for the first time.
-  test: test
+  oauth2Config: {
+    // Step 1 of the OAuth flow; specify where to send the user to authenticate with your API.
+    // Zapier generates the state and redirect_uri, you are responsible for providing the rest.
+    // Note: can also be a function that returns a string
+    authorizeUrl: {
+      url: '{{bundle.inputData.baseUrl}}/oauth/v2/authorize',
+      params: {
+        client_id: '{{bundle.inputData.clientId}}',
+        state: '{{bundle.inputData.state}}',
+        redirect_uri: '{{bundle.inputData.redirect_uri}}',
+        response_type: 'code'
+      }
+    },
+    // Step 2 of the OAuth flow; Exchange a code for an access token.
+    // This could also use the request shorthand.
+    getAccessToken: getAccessToken,
+    // (Optional) If the access token expires after a pre-defined amount of time, you can implement
+    // this method to tell Zapier how to refresh it.
```

Comment:
These 2 lines of comment are not suited for production. Maybe go through all the comment and make it relevant. I think they were copy-pasted from some tutorial.